streaming/src/test/scala/org/apache/spark/streaming/ReceivedBlockHandlerSuite.scala

```diff
@@ -174,6 +176,130 @@ class ReceivedBlockHandlerSuite
     }
   }
 
+  test("Test Block - count messages") {
+    // Test count with BlockManagedBasedBlockHandler
+    testCountWithBlockManagerBasedBlockHandler(true)
```

Comment:
This method of switching the handlers in tests is also bad because it doesn't generalize to handle a third type of handler. Instead, I think it would have been better to just pass in either `withWriteAheadLogBasedBlockHandler` or `withWriteAheadLogBasedBlockHandler` into some function, since both of those methods are instances of a common function supertype. Higher-order functions are a great language feature and should be used more in test code where appropriate.